common_benchmark_suite/README.md

```diff
@@ -1,3 +1,337 @@
+`scripts/generate_model_artifacts.sh` in each framework folder under
+[comparative_suite](/common_benchmark_suite/openxla/benchmark/comparative_suite)
+to generate artifacts. Note that sometimes it can be tricky to generate exported
+model artifacts. Please read the instructions in the scripts or file issues if
+you run into some problems.
+
+The example below generates artifacts for `EXAMPLE_FP32_PT_BATCH*` models:
+
+```sh
+./common_benchmark_suite/openxla/benchmark/comparative_suite/pt/scripts/generate_model_artifacts.sh 'EXAMPLE_FP32_PT_BATCH.+'
+```
+
+After the artifacts are generated, by default they will be stored at:
+
+```sh
+/tmp/${ARTIFACTS_VERSION}/
+# where ${ARTIFACTS_VERSION} follows the format "${FRAMEWORK}_models_${FRAMEWORK_VERISON}_$(date +'%s')"
+
+# For example:
+# /tmp/pt_models_20230723.908_1690223270/
+```
+
+Now upload the aritfacts to GCS by:
+
+```sh
+gcloud storage cp -r  "/tmp/${ARTIFACTS_VERSION}" "gs://iree-model-artifacts/${FULL_QUALIFIED_FRAMEWORK}"
+
+# "${FULL_QUALIFIED_FRAMEWORK}" currently can be:
+# - jax
+# - pytorch
+# - tensorflow
+
+# For example:
+# gcloud storage cp -r /tmp/pt_models_20230723.908_1690223270 gs://iree-model-artifacts/pytorch
+```
+
+Once the artifacts are in GCS, the field `artifacts_dir_url` in
+`def_types.Model` needs to point to the artifact URL, so tools know where to
+download the assoicated artifacts. The URL format should be:
+
+```sh
+https://storage.googleapis.com/iree-model-artifacts/${FULL_QUALIFIED_FRAMEWORK}/${ARTIFACTS_VERSION}/${MODEL_NAME}
+
+# For example:
+# https://storage.googleapis.com/iree-model-artifacts/pytorch/pt_models_20230723.908_1690223270/EXAMPLE_FP32_PT_BATCH1
+```
+
+We also provide templating helpers to populate `artifacts_dir_url`. See
+`ARTIFACTS_DIR_URL_TEMPLATE` and how it is used in
+[model_definitions.py](/common_benchmark_suite/openxla/benchmark/comparative_suite/pt/model_definitions.py).
+
+In addition, the field `exported_model_types` also needs to be populated for
+compiler-level benchmark tools (e.g. XLA-HLO) to know the supported model
+formats. The list depends on what formats the model is exported to and currently
+should be hard-coded with:
+
+| Model's base framework | `exported_model_types` |
+|------------------------|------------------------|
+| JAX                    | `[def_types.ModelArtifactType.STABLEHLO_MLIR, def_types.ModelArtifactType.XLA_HLO_DUMP]` |
+| PyTorch                | `[def_types.ModelArtifactType.LINALG_MLIR]` |
+| Tensorflow             | `[def_types.ModelArtifactType.STABLEHLO_MLIR, def_types.ModelArtifactType.XLA_HLO_DUMP, def_types.ModelArtifactType.TF_SAVEDMODEL_V2]` |
+
+You can see the example commit
+[e2b3566](https://github.com/openxla/openxla-benchmark/commit/e2b356659bd06039f454c5fac08ecabce4b80924)
+for how to populate these fields.
+
+After that, optionally add the new benchmarks to the benchmark list in one of
+`benchmark_all.sh` under [comparative_benchmark](/comparative_benchmark), so
```

Comment:
Might be useful to add a comment about how if the model is in JAX or Tensorflow, the model can also be added to the `benchmark_all.sh` script in xla-hlo for compiler-level benchmarks.

Reply:
Done. I added comments about xla-hlo benchmarks below and the support of JAX and TensorFlow models at the beginning.